File: VcheckSDKDemoIOS/domain/country_models.kt
package com.vcheck.demo.dev.domain

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

data class CountriesResponse (
    @SerializedName("data")
    val data: List<Country>,
    @SerializedName("error_code")
    var errorCode: Int = 0,
    @SerializedName("message")
    var message: String = "")

data class Country(
    @SerializedName("code")
    val code: String,
    @SerializedName("is_blocked")
    val isBlocked: Boolean)

@Parcelize
data class CountryTO(val name: String, val code: String, val flag: String, val isBlocked : Boolean): Parcelable

File: VcheckSDKDemoIOS/domain/CountryModels.swift
//
//  CountryModels.swift
//  VcheckSDKDemoIOS
//
//  Created by Kirill Kaun on 28.04.2022.
//

import Foundation


struct CountriesResponse: Codable {

  var data      : [Country]? = []
  var errorCode : Int?    = nil
  var message   : String? = nil

  enum CodingKeys: String, CodingKey {

    case data      = "data"
    case errorCode = "error_code"
    case message   = "message"
  
  }

  init(from decoder: Decoder) throws {
    let values = try decoder.container(keyedBy: CodingKeys.self)

    data      = try values.decodeIfPresent([Country].self , forKey: .data      )
    errorCode = try values.decodeIfPresent(Int.self    , forKey: .errorCode )
    message   = try values.decodeIfPresent(String.self , forKey: .message   )
 
  }

  init() {

  }

}


struct Country: Codable {

  var code      : String
  var isBlocked : Bool

  enum CodingKeys: String, CodingKey {

    case code      = "code"
    case isBlocked = "is_blocked"
  
  }

  init(from decoder: Decoder) throws {
    let values = try decoder.container(keyedBy: CodingKeys.self)

    code      = try values.decodeIfPresent(String.self , forKey: .code      )!
    isBlocked = try values.decodeIfPresent(Bool.self   , forKey: .isBlocked )!
 
  }

}
